### PR TITLE
Use runtime context only where we require it

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectModel/ProjectContext.cs [hunk 1]
```diff
@@ -85,6 +85,18 @@ public static ProjectContext Create(string projectPath, NuGetFramework framework
         }
 
         /// <summary>
+        /// Creates a project context based on existing context but using runtime target
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="runtimeIdentifiers"></param>
+        /// <returns></returns>
+
+        public static ProjectContext CreateRuntimeContext(ProjectContext context, IEnumerable<string> runtimeIdentifiers)
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/ProjectContext.cs]: This should be an instance method.

File: src/dotnet/commands/dotnet-build/CompileContext.cs [hunk 2]
```diff
@@ -310,6 +316,12 @@ private bool InvokeCompileOnRootProject()
             args.Add("--configuration");
             args.Add(_args.ConfigValue);
 
+            if (!string.IsNullOrWhiteSpace(_args.RuntimeValue))
```
Comment (davidfowl) [inline src/dotnet/commands/dotnet-build/CompileContext.cs]: Remove this and break the script :smile:

File: src/dotnet/commands/dotnet-build/CompileContext.cs [hunk 3]
```diff
@@ -290,6 +290,12 @@ private bool InvokeCompileOnDependency(ProjectDescription projectDependency)
             args.Add(_args.ConfigValue);
             args.Add(projectDependency.Project.ProjectDirectory);
 
+            if (!string.IsNullOrWhiteSpace(_args.RuntimeValue))
```
Comment (davidfowl) [inline src/dotnet/commands/dotnet-build/CompileContext.cs]: Remove it.

**Comment by pakrym** [comment]
@dotnet-bot test this please

**Pull request merged**